### PR TITLE
chore: upgrade to react 17

### PR DESCRIPTION
**Pull request by StevenLeeTW**
upgrade to react 17
peerDependencies support 16 and 17

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
     "build:bundle": "rollup -c ./config/rollup.config.js",
     "build:type": "tsc --project ./config/tsconfig.type.json",
     "lint": "eslint './src/**/*.{ts,tsx}'",
-    "prettier": "prettier --check ./**/*.{ts,tsx,js,json}"
+    "prettier": "prettier --check ./**/*.{ts,tsx,js,json} ./**.js"
```
Comment (danhuang1202) [inline package.json]: Why do you add another for JavaScript files here?

Comment (StevenLeeTW) [reply package.json]: do prettier check babel.config.js, .eslintrc.js ... etc

**Review by danhuang1202 — APPROVED** [review]
👍

**Pull request merged**